cluster-autoscaler/simulator/drainability/rules/customcontroller/rule.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package customcontroller
+
+import (
+	"fmt"
+
+	apiv1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/autoscaler/cluster-autoscaler/simulator/drainability"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/drain"
+	pod_util "k8s.io/autoscaler/cluster-autoscaler/utils/pod"
+)
+
+// Rule is a drainability rule on how to handle replicated pods.
+type Rule struct {
+	minReplicaCount int
+}
+
+// New creates a new Rule.
+func New(minReplicaCount int) *Rule {
+	return &Rule{
+		minReplicaCount: minReplicaCount,
+	}
+}
+
+// Drainable decides what to do with replicated pods on node drain.
+func (r *Rule) Drainable(drainCtx *drainability.DrainContext, pod *apiv1.Pod) drainability.Status {
+	if drainCtx.Listers == nil {
+		return drainability.NewUndefinedStatus()
+	}
+
+	// For now, owner controller must be in the same namespace as the pod
+	// so OwnerReference doesn't have its own Namespace field.
+	controllerNamespace := pod.Namespace
+
+	controllerRef := drain.ControllerRef(pod)
+	if controllerRef == nil {
+		return drainability.NewUndefinedStatus()
+	}
+	refKind := controllerRef.Kind
+
+	if refKind == "ReplicationController" {
+		rc, err := drainCtx.Listers.ReplicationControllerLister().ReplicationControllers(controllerNamespace).Get(controllerRef.Name)
+		// Assume RC is either gone/missing or has too few replicas configured.
+		if err != nil || rc == nil {
+			return drainability.NewBlockedStatus(drain.ControllerNotFound, fmt.Errorf("replication controller for %s/%s is not available, err: %v", pod.Namespace, pod.Name, err))
+		}
+
+		// TODO: Replace the minReplica check with PDB.
+		if rc.Spec.Replicas != nil && int(*rc.Spec.Replicas) < r.minReplicaCount {
+			return drainability.NewBlockedStatus(drain.MinReplicasReached, fmt.Errorf("replication controller for %s/%s has too few replicas spec: %d min: %d", pod.Namespace, pod.Name, rc.Spec.Replicas, r.minReplicaCount))
+		}
+	} else if pod_util.IsDaemonSetPod(pod) {
+		if refKind == "DaemonSet" {
```

Comment:
This should be `!=`. Might be worth adding a unit test that would catch this...

Reply:
Nice catch. Added tests.